### PR TITLE
Vehicles but fixed

### PR DESCRIPTION
**Pull request by Scruq445**
## About the PR
re-adds vehicles, still not mapped yet

## Why / Balance
space mario kart

## Technical details
original code was so bad, had to burn it all down and start from scratch

## Media

[](https://github.com/user-attachments/assets/7dcb8e82-6d16-453c-90ed-aa45491b29d7)

**Changelog**

:cl:
- add: Nanotrasen has fixed their old vehicle's resistance to station environments, they are still being rolled out to stations but Cargo can order a complimentary ATV

## Summary by CodeRabbit

- **New Features**
  - Introduced a robust vehicle system with appearance management and dynamic movement for improved vehicle interaction.
  - Added new vehicle components for enhanced functionality, including sound actions for horns and sirens.
  - Implemented new vehicle spawners and various vehicle key entities for better gameplay diversity.

- **Enhancements**
  - Expanded localization capabilities for the vehicle slot component.
  - Added sound collections for immersive audio experiences related to vehicle operations.
  - Enhanced tagging system to accommodate new vehicle functionalities.

### CODE REVIEW

File: Content.Shared/Vehicles/SharedVehicleSystem.cs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        {
+            ent.Comp.FirstRun = false;
+            ent.Comp.EngineRunning = false;
+            _appearance.SetData(ent.Owner, VehicleState.Animated, false);
+            _ambientSound.SetAmbience(ent.Owner, false);
+            return;
+        }
+
+        Mount(driver, ent.Owner);
+    }
```
[the remark this answers is not in the document]
Comment (Scruq445) [reply Content.Shared/Vehicles/SharedVehicleSystem.cs:182]: help refactor

**Comment by Scruq445** [comment]
build & test check fail is unrelated

File: Content.Shared/Vehicles/SharedVehicleSystem.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Scruq445) [reply Content.Shared/Vehicles/SharedVehicleSystem.cs:182]: turns out adding the horn and siren were raising the insert event
make it check if the args.container contains their uid and return

**Comment by Scruq445** [comment]
also I am not adding the syndicate segway until secways are mapped, because syndicates having their own segway to run away from slow sec seems unfair

**Review by Piras314 — COMMENTED** [review]
the code and everything seems good from skimming it, but it would be nice if you could test (and if possible show video) of some edge cases e.g. falling down from narcolepsy, stuns, death/crit while sitting in a vehicle and moving around after to see if there are any bugs relating to that

**Comment by Scruq445** [comment]
[](https://github.com/user-attachments/assets/30b86f90-4f93-49ba-86ae-c44d74b78c2d)


[](https://github.com/user-attachments/assets/9e09957d-793d-4199-b92e-fcf7f6c888cc)

**Comment by Piras314** [comment]
@Scruq445 Falling down should probably unbuckle you from the vehicle, especially since seemingly movement doesn't work after waking up from sleep without unbuckling and buckling based on the videos, though I'm not sure.

The acceleration should also be much lower (i.e. it should take a much longer time to speed up to max speed)

**Comment by Scruq445** [comment]
> The acceleration should also be much lower (i.e. it should take a much longer time to speed up to max speed)

should I make this global to all vehicles?

**Comment by Piras314** [comment]
> > The acceleration should also be much lower (i.e. it should take a much longer time to speed up to max speed)
> 
> should I make this global to all vehicles?

yeah probably

**Comment by ItsMeThom** [comment]
Will the Ripley/HonkMech be refactored to use the new vehicle system also?

**Comment by Scruq445** [comment]
I'm gonna continue on this later, sleep deprivation is starting to hit me

**Comment by Scruq445** [comment]
> Will the Ripley/HonkMech be refactored to use the new vehicle system also?

Maybe, but I like how the mech feels right now, if a lot of people want vehicular Mechs back, then maybe I'll try re-adding that component

**Comment by Scruq445** [comment]
I'm currently working on a fix for everything mentioned and the TODO, will commit in sometime

**Comment by Aidenkrz** [comment]
closing this for now, create a new PR with a draft status instead of opening PRs for uncompleted features.
